conans/test/functional/toolchains/apple/test_xcodetoolchain.py

```diff
@@ -83,6 +83,7 @@ def package_info(self):
     sdk_version = "11.3"
     settings = "-s arch=x86_64 -s os.sdk_version={} -s compiler.cppstd={} " \
```

Comment:
Unlike te tests in this PR  https://github.com/conan-io/conan/pull/11746 - this one explicitly sets `-s arch=x86_64`, so it's not falling back on the default profile.

This PR is one of two possible choices: we can reflect the fact that macOS on Apple Silicon can run `x86_64` binaries _just fine_ (provided Rosetta is installed) - and then fix the test such that the newr `conan.tols.build.can_run` is used...

**OR** we could dynamically set `-s arch=` according to the current architecture, and amend the `asserts` further down accordingly.